test/parallel/test-repl-history-navigation.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+// Flags: --expose-internals
+
+const common = require('../common');
+const stream = require('stream');
+const REPL = require('internal/repl');
+const assert = require('assert');
+
+// Create an input stream specialized for testing an array of actions
+class ActionStream extends stream.Stream {
+  run(data) {
+    const _iter = data[Symbol.iterator]();
+    const doAction = () => {
+      const next = _iter.next();
+      if (next.done) {
+        // Close the repl. Note that it must have a clean prompt to do so.
+        setImmediate(() => {
+          this.emit('keypress', '', { ctrl: true, name: 'd' });
+        });
```

Comment:
Is there a specific reason for using `setImmediate` here? Shouldn't it also work synchronously?

Reply:
I guess thats not necessary here, I had borrowed this class from other test files.